tests/test_nmapconnector.py

```diff
@@ -0,0 +1,206 @@
+    """
+    Stop HTTPServer
+    :return:
+    """
+    server_v4.shutdown()
+    server_v6.shutdown()
+
+
+def test_nmapconnector_init() -> None:
+    """
+    Check if we can successfully create new NmapConnector instance
+    :return:
+    """
+    NmapConnector()
+
+
+def test_nmapconnector_init_error() -> None:
+    """
+    Raise NmapConnectorInitError and check output of it
+    :return:
+    """
+    with patch(
+        "grinder.nmapconnector.NmapConnector.__init__",
+        side_effect=NmapConnectorInitError("test"),
+    ):
+        with raises(NmapConnectorInitError) as init_error:
+            NmapConnector()
+        assert "Error occured in Nmap Connector module: test" == str(init_error.value)
+
+
+def test_nmapconnector_scan_ipv4() -> None:
+    """
+    Check if we can successfully scan 127.0.0.1 host
+    :return:
+    """
+    nm.scan(
+        host=NmapTestDefaultValues.HOST4,
+        arguments=DefaultNmapScanValues.ARGUMENTS,
+        ports=str(NmapTestDefaultValues.PORT4),
+    )
+
+    assert (
+        nm.get_results()
+        .get(NmapTestDefaultValues.HOST4)
+        .get("tcp")
+        .get(NmapTestDefaultValues.PORT4, False)
+    )
+
+
+def test_nmapconnector_scan_ipv6() -> None:
+    """
+    Check if we can successfully scan ::1 host
+    :return:
+    """
+    nm.scan(
+        host=NmapTestDefaultValues.HOST6,
+        arguments=DefaultNmapScanValues.ARGUMENTS,
+        ports=str(NmapTestDefaultValues.PORT6),
+    )
+
+    assert (
+        nm.get_results()
+        .get(NmapTestDefaultValues.HOST6)
+        .get("tcp")
+        .get(NmapTestDefaultValues.PORT6, False)
+    )
+
+
+def test_nmapconnector_scan_error() -> None:
+    """
+    Raise NmapConnectorScanError and check output of it
+    :return:
+    """
+    with patch(
+        "grinder.nmapconnector.NmapConnector.scan",
+        side_effect=NmapConnectorScanError("test"),
+    ):
+        with raises(NmapConnectorScanError) as scan_error:
+            nm.scan(host=NmapTestDefaultValues.HOST4)
+        assert "Error occured in Nmap Connector module: test" == str(scan_error.value)
+
+
+def test_nmpaconnector_scan_without_any_args() -> None:
+    """
+    Test NmapConnector scan running without any args
+    :return:
+    """
+    with raises(NmapConnectorScanError) as scan_error:
+        nm.scan(host="", arguments="", ports="", sudo=False)
+    assert "'' does not appear to be an IPv4 or IPv6 address" in str(scan_error.value)
+
+
+def test_nmpaconnector_scan_bad_argument() -> None:
+    """
+    Test NmapConnector scan running with bad arguments
+    :return:
+    """
+    with raises(NmapConnectorScanError) as scan_error:
+        nm.scan(host=NmapTestDefaultValues.HOST4, arguments="--bad-argument")
+    assert "nmap: unrecognized option '--bad-argument'" in str(scan_error.value)
+
+
+def test_nmapconnector_get_results() -> None:
+    """
+    Check if we can successfully get NmapConnector scan results
+    :return:
+    """
+    nm.get_results()
```

Comment:
Also, let's check the default values of `get_results()` - you can create here new NmapConnector temporary instance and check that default value of `get_results()` is `[]`, just in case that this value is not equal to `None`, not `0` and so on, it's important to check default value.

Reply:
Maybe it will be better to check in a separate function, like `test_nmapconnector_get_results_default_values`.

Reply:
So it will be like:
```python
def test_nmapconnector_get_results() -> None:
    ...
def test_nmapconnector_get_results_default_values() -> None:
    empty_nm = NmapConnector()
    assert empty_nm.get_results() == ...
...
```